modules/volume/src/processors/volumeregionstatistics.cpp

```diff
@@ -0,0 +1,402 @@
+
+        const auto defaultRanges =
+            util::make_array<4>([&](auto) -> std::optional<dvec2> { return {}; });
+
+        const auto volumeUnit = axes[0].unit * axes[1].unit * axes[2].unit;
+        const auto sumUnits =
+            util::make_array<4>([&](auto) { return volumeUnit * map.valueAxis.unit; });
+
+        const auto posMin = dvec3{data2dest * dvec4{0.0, 0.0, 0.0, 1.0}};
+        const auto posMax = dvec3{data2dest * dvec4{1.0, 1.0, 1.0, 1.0}};
+        std::array<std::optional<dvec2>, 3> sizeRange = {{dvec2{posMin[0], posMax[0]},
+                                                          dvec2{posMin[1], posMax[1]},
+                                                          dvec2{posMin[2], posMax[2]}}};
+
+        regionVolumes = addColumns(*df, "Volume", nRegions, volumeUnit, {});
+        regionSums =
+            addColumns(*df, channels, "Sum", nRegions, sumUnits, defaultRanges, channelLabels);
+        regionMean =
+            addColumns(*df, channels, "Mean", nRegions, valueUnits, defaultRanges, channelLabels);
+        regionMin =
+            addColumns(*df, channels, "Min", nRegions, valueUnits, defaultRanges, channelLabels);
+        regionMax =
+            addColumns(*df, channels, "Max", nRegions, valueUnits, defaultRanges, channelLabels);
+        regionCenter = addColumns(*df, 3, "Center", nRegions, axesUnits, sizeRange, axesNames);
+        regionCoM = addColumns(*df, channels, 3, "CoM", nRegions, axesUnits, sizeRange,
+                               channelLabels, util::make_span(axesNames));
+    }
+
+    size_t getRegion(size_t index) const {
+        return atlasRep->dispatch<size_t, dispatching::filter::UnsignedIntegerScalars>(
+            [&](auto rep) { return rep->getDataTyped()[index]; });
+    }
+    double getValue(size_t index, size_t channel) const {
+        return volumeRep->dispatch<double, dispatching::filter::All>(
+            [&](auto rep) { return util::glmcomp(rep->getDataTyped()[index], channel); });
+    }
+
+    template <Wrapping wrapX, Wrapping wrapY, Wrapping wrapZ>
+    std::shared_ptr<DataFrame> operator()() const {
+        using TStats = Stats<double, wrapX, wrapY, wrapZ>;
+        std::vector<TStats> stats(nRegions * channels, TStats{dvec3{1.0}});
+
+        const util::IndexMapper2D regionMapper(size2_t{nRegions, channels});
+
+        const util::IndexMapper3D indexMapper(dim);
+        util::forEachVoxel(*volumeRep, [&](const size3_t& pos) {
+            const auto p = indexMapper(pos);
+            for (size_t c = 0; c < channels; ++c) {
+                const auto region = regionMapper(getRegion(p) - 1, c);
+                const auto value = getValue(p, c);
+                const auto dpos = dvec3{index2data * dvec4{pos, 1.0}};
+                stats[region].add(dpos, value);
+            }
+        });
+
+        for (auto&& [i, stat] : util::enumerate(stats)) {
+            const auto [region, c] = regionMapper(i);
```

Comment:
I think that iterating over regions would be quite useful in other places. If we would have an Atlas/Region volume it would be nice to expose such functionality there.

Reply:
Im sure that could be nice, but the regionMapper here is just a regular IndexMapper2D.